onnxscript/function_libs/torch_aten/graph_building.py

```diff
@@ -65,169 +64,217 @@ def onnx_dtype(self):
+    # Construct a dictionary of attributes with their names specified in the function
     # definition
-    attributes = args[len(func_ir.inputs) :]
-    onnx_attrs = {}
+    onnx_attributes = {}
 
     # (1) Some/All attributes are supplied as positional arguments
-    # (2) Some attributes are supplied as kwargs
-    # (3) Some arguments in kwargs are not defined in the onnx function
     attr_name_to_protos = collections.OrderedDict(
-        (attr.name, attr) for attr in func_ir.attr_protos
+        (attr.name, attr) for attr in function_ir.attr_protos
     )
-    assert len(attr_name_to_protos) >= len(attributes)
-    for attr_proto, attr_value in zip(attr_name_to_protos.values(), attributes):
-        onnx_attrs[attr_proto.name] = attr_value
 
+    assert len(function_ir.attr_protos) >= len(attributes_in_args)
+    for attr_proto, attr_value in zip(attr_name_to_protos.values(), attributes_in_args):
+        onnx_attributes[attr_proto.name] = attr_value
+
+    # (2) Some/All attributes are supplied as kwargs
     for key, value in kwargs.items():
+        # (3) Some arguments in kwargs are not defined in the onnx function
         if key not in attr_name_to_protos:
             warnings.warn(f"Attribute '{key}' is not defined in the function definition")
             continue
-        # Fill in the values from kwargs
-        onnx_attrs[key] = value
 
-    # Fill in the default values
+        onnx_attributes[key] = value
+
+    # (4) Fill in the default values from the attr_proto if not supplied by caller
     for key, attr_proto in attr_name_to_protos.items():
-        if key not in onnx_attrs:
-            onnx_attrs[key] = attr_proto.value
-
-    return onnx_inputs, onnx_attrs
-
-
-def _convert_kwargs_for_torchscript(kwargs):
-    encoded = {}
-    for attr_name, attr in kwargs.items():
-        if isinstance(attr, float):
-            attr_name += "_f"
-        elif isinstance(attr, int):
-            attr_name += "_i"
-        elif isinstance(attr, str):
-            attr_name += "_s"
-        elif isinstance(attr, list):
-            if isinstance(attr, float):
-                attr_name += "_f"
-            elif isinstance(attr, int):
-                attr_name += "_i"
-        encoded[attr_name] = attr
-    return encoded
+        if key not in onnx_attributes:
+            onnx_attributes[key] = attr_proto.value
+
+    return onnx_inputs, onnx_attributes
 
 
 def _unwrap_tensor_to_torch_value(
-    value: Union[Dict[str, Any], List]
-) -> Union[Dict[str, Any], List]:
-    # unwrap TorchScriptTensor
+    value: ValidArgumentType
+    | dict[str, ValidArgumentType]
+    | list[ValidArgumentType]
+    | tuple[ValidArgumentType, ...]
+) -> torch.Value | str | int | float | dict[str, torch.Value | str | int | float] | list[
+    torch.Value | str | int | float
+] | tuple[torch.Value | str | int | float, ...]:
+    """Unwrap the TorchScriptTensor to torch.Value.
+
+    Args:
+        value: The value to unwrap.
```

Comment:
The description seems redundant regarding to the function name already explaining everything. But I don't have a better idea. feel ignorable.

Reply:
removed